velox/type/Type.h

```diff
@@ -407,24 +407,32 @@ struct TypeParameter {
   const TypeParameterKind kind;
 
   /// Must be not not null when kind is kType. All other properties should be
-  /// null or unset.
+  /// null or unset (other than rowFieldName).
   const TypePtr type;
 
   /// Must be set when kind is kLongLiteral. All other properties should be null
   /// or unset.
   const std::optional<int64_t> longLiteral;
 
+  /// If this parameter is a child of another parent row type, it can optionally
+  /// have a name, e.g, "id" for `row(id bigint)`. Only set when kind is kType
+  const std::optional<std::string> rowFieldName;
```

Comment:
Hopefully this won't increase memory pressure

Reply:
this is only really used during query/task planning (not during execution), so it shouldn't change things much.